mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/impl/PojoTypeManagerContainer.java

```diff
@@ -39,16 +40,23 @@ public static Builder builder() {
 	private final KeyValueProvider<PojoRawTypeIdentifier<?>, AbstractPojoTypeManager<?, ?>> byExactType;
 	private final KeyValueProvider<PojoRawTypeIdentifier<?>, PojoIndexedTypeManager<?, ?>> indexedByExactType;
 	private final KeyValueProvider<String, AbstractPojoTypeManager<?, ?>> byEntityName;
+	private final KeyValueProvider<String, PojoRawTypeModel<?>> typeModelByEntityName;
+	private final KeyValueProvider<PojoRawTypeIdentifier<?>, Set<? extends AbstractPojoTypeManager<?, ?>>> allByNonInterfaceSuperType;
 
 	private final Map<PojoRawTypeIdentifier<?>, Set<? extends PojoIndexedTypeManager<?, ?>>> indexedBySuperType;
 
 	private final Collection<PojoIndexedTypeManager<?, ?>> allIndexed;
+	private final Set<PojoRawTypeIdentifier<?>> allIndexedAndContainedTypes;
+	private final Set<PojoRawTypeIdentifier<?>> allNonInterfaceSuperTypes;
+	private final Set<Class<?>> allNonInterfaceSuperTypesClasses;
 
 	private PojoTypeManagerContainer(Builder builder) {
 		// Use a LinkedHashMap for deterministic iteration in the "all" set
 		Map<PojoRawTypeIdentifier<?>, AbstractPojoTypeManager<?, ?>> byExactTypeContent = new LinkedHashMap<>();
 		Map<PojoRawTypeIdentifier<?>, PojoIndexedTypeManager<?, ?>> indexedByExactTypeContent = new LinkedHashMap<>();
 		Map<String, AbstractPojoTypeManager<?, ?>> byEntityNameContent = new LinkedHashMap<>();
+		Map<String, PojoRawTypeModel<?>> typeModelByEntityNameContent = new LinkedHashMap<>();
```

Comment:
yes sir 😃 👍🏻

Reply:
Sorry :x But this time it's not just me nitpicking, it has real implications with native images...

Reply:
hey, no worries 😃 I was hesitant to go with the type identifier since, at some point earlier, you mentioned that this thing should provide type managers, so that's why I've called it out in the comment so that you suggest a better thing to save in that map 😉